metadata-ingestion/tests/unit/test_transform_dataset.py

```diff
@@ -2841,3 +2844,344 @@ def test_add_dataset_data_product_transformation():
+            builder.make_owner_urn(user, owner_type=builder.OwnerType.USER)
+        )
+
+    # remove `example.*`
+    config: List[Union[re.Pattern, str]] = [r"@example\.\S*"]
+    expected_user_emails: List[str] = [
+        "ABCDEF:email_id",
+        "ABCDEF:123email_id",
+        "email_id",
+        "email_id",
+        "email_test:XYZ",
+        "email_id:id1",
+        "email_id:id2",
+    ]
+    expected_owner_urns: List[str] = []
+    for user in expected_user_emails:
+        expected_owner_urns.append(
+            builder.make_owner_urn(user, owner_type=builder.OwnerType.USER)
+        )
+    _test_clean_owner_urns(pipeline_context, in_owner_urns, config, expected_owner_urns)
+
+
+def test_clean_owner_urn_transformation_remove_capital_letters(mock_datahub_graph):
+    pipeline_context = PipelineContext(run_id="transformer_pipe_line")
+    pipeline_context.graph = mock_datahub_graph(DatahubClientConfig())
+
+    user_emails = [
+        "ABCDEF:email_id@example.com",
+        "ABCDEF:123email_id@example.com",
+        "email_id@example.co.in",
+        "email_id@example.co.uk",
+        "email_test:XYZ@example.com",
+        "email_id:id1@example.com",
+        "email_id:id2@example.com",
+    ]
+
+    in_owner_urns: List[str] = []
+    for user in user_emails:
+        in_owner_urns.append(
+            builder.make_owner_urn(user, owner_type=builder.OwnerType.USER)
+        )
+
+    # remove any CAPITAL chars between `:` and `@`
+    config: List[Union[re.Pattern, str]] = ["(?<=:)[A-Z]+(?=@)"]
+    expected_user_emails: List[str] = [
+        "ABCDEF:email_id@example.com",
+        "ABCDEF:123email_id@example.com",
+        "email_id@example.co.in",
+        "email_id@example.co.uk",
+        "email_test:@example.com",
+        "email_id:id1@example.com",
+        "email_id:id2@example.com",
+    ]
+    expected_owner_urns: List[str] = []
+    for user in expected_user_emails:
+        expected_owner_urns.append(
+            builder.make_owner_urn(user, owner_type=builder.OwnerType.USER)
+        )
+    _test_clean_owner_urns(pipeline_context, in_owner_urns, config, expected_owner_urns)
+
+
+def test_clean_owner_urn_transformation_remove_pattern_with_alphanumeric_value(
+    mock_datahub_graph,
+):
+    pipeline_context = PipelineContext(run_id="transformer_pipe_line")
+    pipeline_context.graph = mock_datahub_graph(DatahubClientConfig())
+
+    user_emails = [
+        "ABCDEF:email_id@example.com",
+        "ABCDEF:123email_id@example.com",
+        "email_id@example.co.in",
+        "email_id@example.co.uk",
+        "email_test:XYZ@example.com",
+        "email_id:id1@example.com",
+        "email_id:id2@example.com",
+    ]
+
+    in_owner_urns: List[str] = []
+    for user in user_emails:
+        in_owner_urns.append(
+            builder.make_owner_urn(user, owner_type=builder.OwnerType.USER)
+        )
+
+    # remove any pattern having `id` followed by any digits
+    config: List[Union[re.Pattern, str]] = [r"id\d+"]
```

Comment:
Based on this I would expect `id` and digits to be removed. Why is that not the case? What decides only the digits are going to be removed?

Reply:
current RegEx will look for any string which is having `id` and then `digits`.

input values:
```
 "email_id:id1@example.com",
 "email_id:id2@example.com",
```

here, we are removing `id1` and `id2`... `_id` remains there as there are no digits following.

and hence the expected values are 

```
"email_id:@example.com",
"email_id:@example.com",
```